pkg/apis/serving/v1alpha2/inferenceservice_types.go

```diff
@@ -60,6 +60,29 @@ type DeploymentSpec struct {
 	// This is the up bound for autoscaler to scale to
 	// +optional
 	MaxReplicas int `json:"maxReplicas,omitempty"`
+	// Activate request/response logging
+	// +optional
+	InferenceLogger *InferenceLogger `json:"inferenceLogger,omitempty"`
+}
+
+type InferenceLoggerType string
+
+const (
+	InferenceLogBoth     InferenceLoggerType = "both"
+	InferenceLogRequest  InferenceLoggerType = "request"
+	InferenceLogresponse InferenceLoggerType = "response"
+)
+
+type InferenceLogger struct {
+	// URL to send request logging CloudEvents
+	// +optional
+	Url *string `json:"url,omitempty"`
+	// What payloads to log
+	// +optional
+	LogType *InferenceLoggerType `json:"logType,omitempty"`
+	// What percentage of requests to log, value between 0->1
+	// +optional
+	Sample *float32 `json:"sample,omitempty"`
```

Comment:
Should we extend this to be a struct?
```
logger:
  type: "request"
  sample: 
    percent: 10
```

In the future we can sample based off a key, etc.

Reply:
Can do. Other option is to remove and leave for future extension.

Reply:
Yes! Let's punt.